core/src/main/scala/org/apache/spark/h2o/converters/SparkDataFrameConverter.scala

```diff
@@ -78,12 +86,12 @@ trait SparkDataFrameConverter extends Logging with ConverterUtils {
   def perSQLPartition(types: Seq[(Seq[Int], StructField, Byte)])
                      (keyName: String, vecTypes: Array[Byte], uploadPlan: Option[immutable.Map[Int, NodeDesc]])
                      (context: TaskContext, it: Iterator[Row]): (Int, Long) = {
+    val asArr = it.toArray[Row] // need to buffer the iterator in order to get its length
```

Comment:
Iterator are passed because it's Spark api we have to play with. I need to buffer the array in advance because I need to get the number of elements before the iteration over them which is not possible with iterator. 

Agree that array is not necessary, can change to `List`